feat(el): greek calendar & locale

ciesiolka opened the pull request:
This commit contains:
- translation of saint names as appear in Roman Missal in Greek language (2006)
- celebrations from proper calendars of Greek Dioceses of Athens, Syros and Tinos (all from 2013)
- saints mentioned in those calendars were added to `martyrology.ts`, except for:
  - dedications of own churches
  - `timothy_of_ephesus_and_titus_the_first_bishop_of_crete_bishops` - variant of existing `timothy_of_ephesus_and_titus_of_crete_bishops`
  - `lydia_of_philippi_first_european_christian` - variant of `lydia_of_philippi`
  - `zechariah_and_elizabeth_parents_of_john_the_baptist`
  - `twelve_apostles`

Only some of the inserted saints were translated to Latin. To be fixed in a separate PR.

Names of some of saints inserted to the General Calendar after 2006 were added. If found, please report missing ones.

Important notice, not directly related to the commit: The Easter in Greece is celebrated according to Julian calendar, in the same date as in Orthodox Church. It means the Ash wednesday, Palm Sunday, Pentecost, Corpus Christi, etc. are shifted. I don't know if romcal supports that.

ciesiolka committed:
feat(el): january saints

ciesiolka committed:
feat(el): february

ciesiolka committed:
feat(el): march saints

ciesiolka committed:
feat(el): april and may

ciesiolka committed:
feat(el):  june

ciesiolka committed:
feat(el): july

ciesiolka committed:
feat(el): august

ciesiolka committed:
feat(el): september

ciesiolka committed:
feat(la): latin translations

ciesiolka committed:
feat(el): seasons

ciesiolka committed:
feat(el): martyrology

ciesiolka committed:
feat(en): added english translation of missing names

ciesiolka committed:
feat(el): added missing saints from general calendar

ciesiolka committed:
enh(el): martyrology

ciesiolka committed:
feat(el): unified letter case

tukusejssirs commented:
Hey, @ciesiolka, for awesome contribition! 🙏

I’ll try and review the PR today, however, it’d be awesome if you could provide us the source (-s) (just take some pictures of the selected pages + front cover, title case and title case verso, even privately if you don’t want to publish it to everyone). Thanks in advance! 🙏

ciesiolka commented:
> Hey, @ciesiolka, for awesome contribition! 🙏
> 
> I’ll try and review the PR today, however, it’d be awesome if you could provide us the source (-s) (just take some pictures of the selected pages + front cover, title case and title case verso, even privately if you don’t want to publish it to everyone). Thanks in advance! 🙏

Can you share with me a direct contact to you? Are you using like Discord? If not, e-mail would be fine.

tukusejssirs commented:
> Can you share with me a direct contact to you? Are you using like Discord? If not, e-mail would be fine.

I prefer using email, but use Telegram or Diacord if you wish. See my updated [profile](https://github.com/tukusejssirs).

Thanks! :pray:

emagnier commented:
Thank again @ciesiolka for this great contribution!!

> Important notice, not directly related to the commit: The Easter in Greece is celebrated according to Julian calendar, in the same date as in Orthodox Church. It means the Ash wednesday, Palm Sunday, Pentecost, Corpus Christi, etc. are shifted. I don't know if romcal supports that.

Currently, Romcal only support Easter calculation within a Gregorian calendar. But calculating Easter according to Julian calendar is definitively something that we can add to Romcal!

tukusejssirs commented on lib/particular-calendars/costa-rica.ts:
```diff
@@ -13,5 +13,10 @@ export class CostaRica extends CalendarDef {
       dateDef: { dateFn: 'pentecostSunday', addDay: 4 },
       properCycle: ProperCycles.ProperOfTime,
     },
+
+    our_lady_of_angels: {
+      precedence: Precedences.GeneralSolemnity_3,
+      dateDef: { month: 8, date: 2 },
+    },
```
IMHO, adding `our_lady_of_angels` to `costa-rica.ts` is out of scope of this PR. Consider moving it to as separate PR.

TobiTenno replied:
moved to #546

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,501 @@
+    martin_of_tours_bishop: 'Αγίου Μαρτίνου, επισκόπου',
+    mary_magdalene: 'Αγίας Μαρίας της Μαγδαληνής',
+    mary_magdalene_de_pazzi_virgin: 'Αγίας Μαρίας Μαγδαληνής de’ Pazzi, παρθένου',
+    mary_mother_of_god: 'Tης Υπεραγίας Θεοτόκου Μαρίας',
```
You are missing the translation of `The Octave Day of the Nativity of the Lord` to Greek.

Note that I am not sure if they use colon (`:`) in Greek, as I haven’t found that character on Greek keyboard layout. Do change it if you find a better punctuation mark.

Also note that `'Tης' !== 'Της'`, as the former uses `T` from Latin alphabet and the latter `Τ` from Greek alphabet.

```suggestion
    mary_mother_of_god: 'Ογδοάδα των Χριστουγέννων: Της Υπεραγίας Θεοτόκου Μαρίας',
```

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,501 @@
+    },
+
+    ordinary_time: {
+      season: 'Kοινή περίοδος του έτους',
```
I believe you used Latin `K` instead of Greek `Κ`.

```suggestion
      season: 'Κοινή περίοδος του έτους',
```

tukusejssirs commented:
@ciesiolka, @emagnier, I have started the review of this PR, but it has some issues that needs to be fixed before merging, therefore I have just converted it to a draft PR.

I want to go through all changes from all points of view (`martyrology.ts`, celebrations, localisation, grammar) before I publish the required changes. I hope to do so today.

While there are some big issues, the PR itslef is done very well. I consider the main issue the missing contribution docs, which I wanted to create long time ago (with @emagnier’s approval), therefore this is my fault.

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,501 @@
+    ordinary_time: {
+      season: 'Kοινή περίοδος του έτους',
+      weekday:
+        '$t(weekdays:{{dow}}) της $t(ordinals:{{week}}, { "context": "genitive" }) της κοινής περίοδου του έτους',
```
My spellchecker suggests to use either `περίοδο` or `περίοδοι`. It could be wrong, however, I want to verify it.

tukusejssirs replied:
While I have access to an actual source of `ordinary_time.weekday`, I have updated this to use `περιόδου` just like in `ordinary_time.sunday` for which I have a source (although I reference it from p611 of `mr_el_2006_ed3`, the one I am talking about is written with diacritics on the last line of [p41](https://drive.google.com/file/d/1h05-fjMkOOytpqFuM6ygPIFzwcdofZFk/view?usp=drive_link)).

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,501 @@
+      season: 'Kοινή περίοδος του έτους',
+      weekday:
+        '$t(weekdays:{{dow}}) της $t(ordinals:{{week}}, { "context": "genitive" }) της κοινής περίοδου του έτους',
+      sunday: '$t(ordinals:{{week}}) Κυριακή της κοινής περίοδου του έτους',
```
My spellchecker suggests to use either `περίοδο` or `περίοδοι`. It could be wrong, however, I want to verify it.

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,501 @@
+      holy_week_day: 'Μεγάλη $t(weekdays:{{dow}})',
+    },
+
+    paschal_triduum: { season: 'Πασχαλινό τριήμερο' },
```
Incorrect characters (`ό` and `ή`).

```suggestion
    paschal_triduum: { season: 'Πασχαλινό τριήμερο' },
```

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,501 @@
+    paschal_triduum: { season: 'Πασχαλινό τριήμερο' },
+
+    easter_time: {
+      season: 'Περίοδος του Πάσχα',
```
Incorrect characters (`ί` and `ά`).

```suggestion
      season: 'Περίοδος του Πάσχα',
```

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,501 @@
+      season: 'Περίοδος του Πάσχα',
+      weekday: '$t(weekdays:{{dow}}) της $t(ordinals:{{week}}, { "context": "abbreviation" }) Εβδομάδας του Πάσχα',
+      sunday: '$t(ordinals:{{week}}) Κυριακή του Πάσχα',
+      octave: '$t(weekdays:{{dow}}) της Διακαινησίμου',
```
My spell-checker suggests to change this to use `Διακαινήσιμος`. It could be wrong, however, I want to verify it.

tukusejssirs replied:
It is actually correct, however, I am not 100 % about the diacritics (as in Greek, the diacritics is not printed/added to uppercase letters; [src (private)](https://drive.google.com/file/d/10gRyDBAgmyEL6VBHmU2lxOnoK0thaXjB/view?usp=drive_link)). Also note that I have no idea what the page number is, as it is not printed on the pages with headings of parts and section level in `mr_el_2006_ed3`.

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,501 @@
+  },
+
+  months: {
+    0: 'Ιανουάριου',
```
[Wiktionary](https://en.wiktionary.org/wiki/%CE%99%CE%B1%CE%BD%CE%BF%CF%85%CE%B1%CF%81%CE%AF%CE%BF%CF%85) suggests to use `Ιανουαρίου` as Genitive. I presume we need Genitive, however, we could consider to add Nominative too.

Also compare this to `Κυριακή μετά τις 6 Ιανουαρίου` which can be found int `mr_el_2006_ed3` in the Calendar (Γενικο Πωμαϊκο εορτολογιο Ιανουαριος) in the bottom of the page.

```suggestion
    0: 'Ιανουαρίου',
```

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,501 @@
+
+  months: {
+    0: 'Ιανουάριου',
+    1: 'Φεβρουάριου',
```
[Wiktionary](https://en.wiktionary.org/wiki/%CE%A6%CE%B5%CE%B2%CF%81%CE%BF%CF%85%CE%B1%CF%81%CE%AF%CE%BF%CF%85#Greek) suggests to use `Φεβρουαρίου` as Genitive.

```suggestion
    1: 'Φεβρουαρίου',
```

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,501 @@
+  months: {
+    0: 'Ιανουάριου',
+    1: 'Φεβρουάριου',
+    2: 'Μάρτιου',
```
[Wiktionary](https://en.wiktionary.org/wiki/%CE%9C%CE%B1%CF%81%CF%84%CE%AF%CE%BF%CF%85#Greek) suggests to us `Μαρτίου` as Genitive.

```suggestion
    2: 'Μαρτίου',
```

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,501 @@
+    0: 'Ιανουάριου',
+    1: 'Φεβρουάριου',
+    2: 'Μάρτιου',
+    3: 'Απρίλιου',
```
[Wiktionary](https://en.wiktionary.org/wiki/%CE%91%CF%80%CF%81%CE%B9%CE%BB%CE%AF%CE%BF%CF%85#Greek) suggests to use `Απριλίου` as Genitive.

```suggestion
    3: 'Απριλίου',
```

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,501 @@
+    1: 'Φεβρουάριου',
+    2: 'Μάρτιου',
+    3: 'Απρίλιου',
+    4: 'Μάιου',
```
[Wiktionary](https://en.wiktionary.org/wiki/%CE%9C%CE%B1%CE%90%CE%BF%CF%85#Greek) suggests to use `Μαΐου` as Genitive.

```suggestion
    4: 'Μαΐου',
```

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,501 @@
+    2: 'Μάρτιου',
+    3: 'Απρίλιου',
+    4: 'Μάιου',
+    5: 'Ιούνιου',
```
[Wiktionary](https://en.wiktionary.org/wiki/%CE%99%CE%BF%CF%85%CE%BD%CE%AF%CE%BF%CF%85#Greek) suggests to use `Ιουνίου` as Genitive.

```suggestion
    5: 'Ιουνίου',
```

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,501 @@
+    3: 'Απρίλιου',
+    4: 'Μάιου',
+    5: 'Ιούνιου',
+    6: 'Ιούλιου',
```
[Wiktionary}(https://en.wiktionary.org/wiki/%CE%99%CE%BF%CF%85%CE%BB%CE%AF%CE%BF%CF%85#Greek) suggests to use `Ιουλίου` as Genitive.

```suggestion
    6: 'Ιουλίου',
```

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,501 @@
+    4: 'Μάιου',
+    5: 'Ιούνιου',
+    6: 'Ιούλιου',
+    7: 'Αύγουστου',
```
[Wiktionary](https://en.wiktionary.org/wiki/%CE%91%CF%85%CE%B3%CE%BF%CF%8D%CF%83%CF%84%CE%BF%CF%85#Greek) suggests to use `Αυγούστου` as Genitive.

```suggestion
    7: 'Αυγούστου',
```

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,501 @@
+    5: 'Ιούνιου',
+    6: 'Ιούλιου',
+    7: 'Αύγουστου',
+    8: 'Σεπτέμβριου',
```
[Wiktionary](https://en.wiktionary.org/wiki/%CE%A3%CE%B5%CF%80%CF%84%CE%B5%CE%BC%CE%B2%CF%81%CE%AF%CE%BF%CF%85#Greek) suggests to use `Σεπτεμβρίου` as Genitive.

```suggestion
    8: 'Σεπτεμβρίου',
```

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,501 @@
+    6: 'Ιούλιου',
+    7: 'Αύγουστου',
+    8: 'Σεπτέμβριου',
+    9: 'Οκτώβριου',
```
[Wiktionary](https://en.wiktionary.org/wiki/%CE%9F%CE%BA%CF%84%CF%89%CE%B2%CF%81%CE%AF%CE%BF%CF%85#Greek) suggests to use `Οκτωβρίου` as Genitive.

```suggestion
    9: 'Οκτωβρίου',
```

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,501 @@
+    7: 'Αύγουστου',
+    8: 'Σεπτέμβριου',
+    9: 'Οκτώβριου',
+    10: 'Νοέμβριου',
```
[Wiktionary](https://en.wiktionary.org/wiki/%CE%9D%CE%BF%CE%B5%CE%BC%CE%B2%CF%81%CE%AF%CE%BF%CF%85#Greek) suggests to use `Νοεμβρίου` as Genitive.

```suggestion
    10: 'Νοεμβρίου',
```

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,501 @@
+    8: 'Σεπτέμβριου',
+    9: 'Οκτώβριου',
+    10: 'Νοέμβριου',
+    11: 'Δεκέμβριου',
```
[Wiktionary](https://en.wiktionary.org/wiki/%CE%94%CE%B5%CE%BA%CE%B5%CE%BC%CE%B2%CF%81%CE%AF%CE%BF%CF%85#Greek) suggests to use `Δεκεμβρίου` as Genitive.

```suggestion
    11: 'Δεκεμβρίου',
```

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,501 @@
+    fidelis_of_sigmaringen_priest: 'Αγίου Φιδελίου de Sigmaringen, πρεσβυτέρου και μάρτυρος',
+    first_martyrs_of_the_holy_roman_church: 'Αγίων Πρωτομαρτύρων της Εκκλησίας της Ρώμης',
+    frances_of_rome_religious: 'Αγίας Φρανκίσκας της Ρωμαίας, μοναχής',
+    francis_de_sales_bishop: 'Αγίου Φρανκίσκου de Sales, επισκόπου και διδασκάλου της Εκκλησίας',
```
```suggestion
    francis_de_sales_bishop: 'Αγίου Φραγκίσκου de Sales, επισκόπου και διδασκάλου της Εκκλησίας',
```

tukusejssirs committed:
fix(el): verify all celebrations listed in the GRC in `mr_el_2006_ed3`

Review of #459

tukusejssirs reviewed (changes requested):
@ciesiolka, my notes regarding this PR:

- According to Wikipedia, there are the following dioceses in Greece (correct me if I’m wrong; cf [catholic-hierarchy.org](https://www.catholic-hierarchy.org/country/dgr2.html)):
  - list:
    - Archdiocese of Athens;
    - Archdiocese of Rhodos (or Rhodes?);
    - Archdiocese of Corfù, Zante and Cefalonia (or `Corfu`?);
    - Archdiocese of Naxos, Andros, Tinos and Mykonos;
    - Diocese of Chios (or Scio?);
    - Diocese of Crete;
    - Diocese of Santorini (or Thira?);
    - Diocese of Syros and Milos;
    - Apostolic Vicariate of Thessaloniki;
  - notes:
    - You could add the missing diocesan calendars if you want, possibly in a separate PR.
    - There is no _Diocese of Athens_, as it isi _Archdiocese of Athens_.
    - As I understand it (based on the sources linked above), Dioceses Syros and Tinos are merged into _Diocese of Syros and Milos_ and _Archdiocese of Naxos, Andros, Tinos and Mykonos_. That said, it might be that each of these (sub-) dioceses (which, as I understand it, were merged at some time) still has some celebrations that are celebrated on their territory only (and on the territory of the entire, merged diocese).

---

The TODO list is an unordered list of tasks which need to be done, either in this PR or in a follow-up PRs. I will mark them as completed, as I go through the list. Some of the change requests and comments are being added to my [fork of the `greek-locale` branch](https://github.com/tukusejssirs/romcal/tree/greek-locale).

<details>
  <summary>TODO</summary>

- [x] add code comments after each l10n line with the source of the string (e.g. `src: mr_el_2006_ed3` or `based on: mr_el_2006_ed3`);
- [ ] verify additions to `martyrology.ts`;
- [ ] consider adding the following celebrations:
   - [ ] dedications of own churches;
   - [ ] `timothy_of_ephesus_and_titus_the_first_bishop_of_crete_bishops` - variant of existing `timothy_of_ephesus_and_titus_of_crete_bishops`:
     - consider replacing the current key name (and possibly key values in certain locales) to the new key;
   - [ ] `lydia_of_philippi_first_european_christian` - variant of lydia_of_philippi;
   - [ ] `zechariah_and_elizabeth_parents_of_john_the_baptist`;
   - [ ] `twelve_apostles`;
- [ ] localise all added celebrations to all locales (this might be fixed in a separate PR);
- [ ] verify if all celebrations of the Greek Proper (based on the calendar found in `mr_el_2006_ed3`) are correctly defined in `.names` in `el.ts` (pay heed to potential celebration transfers or removals in comparison with the GRC):
   - [ ] l10n:
      - [x] January;
      - [x] February;
      - [x] March;
      - [x] April;
      - [x] May;
      - [x] June;
      - [x] July;
      - [x] August;
      - [x] September;
      - [x] October;
      - [x] November;
      - [x] December;
      - [x] other pages shared with us;
      - [ ] `names.dedication_of_consecrated_churches`: find the page number in `mr_el_2006_ed3`;
   - [ ] celebration metadata (date, rank, transfer, titles):
      - [x] January;
      - [x] February;
      - [x] March;
      - [x] April;
      - [x] May;
      - [x] June;
      - [x] July;
      - [x] August;
      - [x] September;
      - [x] October;
      - [x] November;
      - [x] December;
      - [ ] other pages shared with us;
- [ ] sort the items in `martyrology.ts` by key name;
- [ ] verify existing l10n key values to `el`;
- [ ] verify l10n keys in `.seasons`;
- [ ] verify l10n keys in `.ranks`;
- [ ] verify l10n keys in `.cycles`;
- [ ] verify l10n keys in `.weekdays`;
- [ ] verify l10n keys in `.months`;
- [ ] verify l10n keys in `.ordinals`;
- [ ] verify l10n keys in `.colors`;
- [ ] spell-check l10n keys in `.names`;
- [x] spell-check l10n keys in other parts of `el.ts` files (except for `.names`);
- [ ] `martyrology.ts` additions verification:
   - [x] `agapitus_of_palestrina_martyr`;
   - [ ] `christopher_of_anatolia_martyr`:
      - is he not the same as `christopher_of_palestine_martyr`? if so, should we rename the old key to the new one?
   - [ ] `cyriacus_martyr`:
      - consider changing the name to `Cyriac`;
      - add surname:
        - what should it be?
        - he was buried at the seventh milestone of the _Via Ostiensis_ → `Saint Cyriacus of the Via Ostiensis`;
        - the Wikipedia article mentions him once as `Saint Cyriacus at the Baths (of Diocletian)` → `Saint Cyriacus of the Baths (of Diocletian)`;
        - at another place, the Wikipedia article says there is a chuched titled as `Church of Saint Cyriacus in the Baths of Diocletian` → `Saint Cyriacus at the Baths of Diocletian` (`la`: `Sanctus Ciriacus in Thermis Diocletiani`; `it`: `S. Ciriaco alle Terme di Diocleziano`);
      - if he really was a deacon (or if he is really venerated/celebrated as such), therefore change his name to `cyriacus_${surname}_deacon`; otherwise keep the `martyr` title in the key name;
   - [ ] `emilie_de_vialar_virgin`:
      - add this to `martyrology.ts`, as it is missing from it;
      - rename her to `emily` in English and thus the key name;
      - consider changing the key name to include `Anne Marguerite Adelaide Emily de Vialar`;
   - [ ] `lawrence_justinian`:
      - add `bishop` title to the key name;
   - [ ] `mammes_of_caesarea_martyr`:
      - add `approx` to `.dateOfDeath`;
   - [ ] `marcellinus_of_ancona_bishop`:
      - add `approx` to `.dateOfDeath`;
   - [ ] `mary_mother_of_hope`:
      - consider using `Our Lady` in `en` instead of `Blessed Virgin Mary`;
   - [x] `miltiades_pope`;
   - [ ] `our_lady_of_angels`;
   - [ ] `our_lady_of_faneromeni`:
      - rename to `our_lady_of_lefkada`;
   - [ ] `pelagia_the_penitent_virgin`:
      - if she is St Pelagia the Penitent/Harlot (AKA St Pelagia of Antioch), she was not a virgin;
      - she could be [St Pelagia the Virgin](https://en.wikipedia.org/wiki/Pelagia_the_Virgin) (AKA St Pelagia of Antioch), but then she is not St Pelagia the Penitent;
   - [ ] `romanus_martyr`:
      - add surname: `Ostiarius`;
   - [ ] `thecla_of_iconium_virgin`:
      - add `.dateOfDeath`: `1st century AD`;
   - [ ] `theodore_stratelates`;
   - [ ] `timothy_of_ephesus_and_titus_the_first_bishop_of_crete_bishops`:
      - `timothy_of_ephesus_and_titus_the_first_bishop_of_crete_bishops: 'Saint Titus, the first Bishop of Crete, Saint Timothy, Bishops',`;
      - remove this key in favour of `timothy_of_ephesus_and_titus_of_crete_bishops`, as it is the same celebration;
      - prefer the old key, as in English, it is used like that and no `the First Bishop of ${dioceseName}` is used;
- [ ] additions l10n verification (incl surnames and a single/main title in key names):
   - [ ] `agapitus_of_palestrina_martyr` ([src](https://en.wikipedia.org/wiki/Agapitus_of_Palestrina)):
      - `agapitus_of_palestrina_martyr: 'Saint Agapitus, Martyr'`;
      - [ ] `cs`;
      - [ ] `es`;
      - [ ] `fr`;
      - [ ] `hu`;
      - [ ] `it`;
      - [ ] `la`;
      - [ ] `pl`;
      - [ ] `pt-br`;
      - [ ] `sk`;
      - consider adding the surname to the key values and to `.name` in `martyrology.ts`;
   - [ ] `christopher_of_anatolia_martyr` ([src](https://en.wikipedia.org/wiki/christopher)):
      - `christopher_of_anatolia_martyr: 'Saint Christopher, Martyr',`;
      - [ ] `cs`;
      - [ ] `es`;
      - [ ] `fr`;
      - [ ] `hu`;
      - [ ] `it`;
      - [ ] `la`;
      - [ ] `pl`;
      - [ ] `pt-br`;
      - [ ] `sk`;
   - [ ] `cyriacus_martyr` ([src](https://en.wikipedia.org/wiki/Cyriacus)):
      - `cyriacus_martyr: 'Saint Cyriacus, Martyr',`;
      - [ ] `cs`;
      - [ ] `es`;
      - [ ] `fr`;
      - [ ] `hu`;
      - [ ] `it`;
      - [ ] `la`;
      - [ ] `pl`;
      - [ ] `pt-br`;
      - [ ] `sk`;
   - [ ] `dedication_of_the_basilica_of_our_lady_of_the_snows_tinos_grece`:
      - `dedication_of_the_basilica_of_our_lady_of_the_snows_tinos_grece: 'The Dedication of the Basilica of Our Lady of the Snows in Tinos, Grece', // aka. Salus Populi Romani`;
      - fix typo: `Grece` → `Greece`;
      - consider adding this to `martyrology.ts`;
      - [ ] `cs`;
      - [ ] `es`;
      - [ ] `fr`;
      - [ ] `hu`;
      - [ ] `it`;
      - [ ] `la`;
      - [ ] `pl`;
      - [ ] `pt-br`;
      - [ ] `sk`;
   - [ ] `dedication_of_the_cathedral_of_saint_dionysius_athens`:
      - `dedication_of_the_cathedral_of_saint_dionysius_athens: 'The Dedication of the Basilica of Saint Dionysius in Athens, Greece',`;
      - consider adding this to `martyrology.ts`;
      - [ ] `cs`;
      - [ ] `es`;
      - [ ] `fr`;
      - [ ] `hu`;
      - [ ] `it`;
      - [ ] `la`;
      - [ ] `pl`;
      - [ ] `pt-br`;
      - [ ] `sk`;
   - [ ] `emilie_de_vialar_virgin`:
      - `emilie_de_vialar_virgin: 'Saint Emily de Vialar',`;
      - [ ] `cs`;
      - [ ] `es`;
      - [ ] `fr`;
      - [ ] `hu`;
      - [ ] `it`;
      - [ ] `la`: `S. Æmilia de Vialar, virginis`;
      - [ ] `pl`;
      - [ ] `pt-br`;
      - [ ] `sk`;
   - [ ] `lawrence_justinian` ([src](https://en.wikipedia.org/wiki/Lawrence_Justinian)):
      - `lawrence_justinian: 'Saint Lawrence Justinian, bishop',`;
      - [ ] `cs`;
      - [ ] `es`;
      - [ ] `fr`;
      - [ ] `hu`;
      - [ ] `it`;
      - [ ] `la`;
      - [ ] `pl`;
      - [ ] `pt-br`;
      - [ ] `sk`;
   - [ ] `lydia_of_philippi_first_european_christian`:
      - `lydia_of_philippi_first_european_christian: 'Saint Lydia of Philippi, the first Christian of Europe',`;
      - [ ] `cs`;
      - [ ] `es`;
      - [ ] `fr`;
      - [ ] `hu`;
      - [ ] `it`;
      - [ ] `la`;
      - [ ] `pl`;
      - [ ] `pt-br`;
      - [ ] `sk`;
   - [ ] `mammes_of_caesarea_martyr` ([src](https://en.wikipedia.org/wiki/Mammes_of_Caesarea)):
      - add missing title to `en` l10n value;
      - [ ] `cs`;
      - [ ] `es`;
      - [ ] `fr`;
      - [ ] `hu`;
      - [ ] `it`;
      - [ ] `la`;
      - [ ] `pl`;
      - [ ] `pt-br`;
      - [ ] `sk`;
   - [ ] `marcellinus_of_ancona_bishop` ([src](https://catholic.net/op/articles/1654/cat/1205/saint-marcellinus-of-ancona.html));
   - [ ] `martha_of_bethany`:
      - `martha_of_bethany: 'Saint Martha',`;
      - check how is this key actually used, as it was updated by Pope Francis to include St Lazarus and St Mary;
      - if it is actually used, I presume it is used with a patronage or titular title;
      - [ ] `cs`;
      - [ ] `es`;
      - [ ] `fr`;
      - [ ] `hu`;
      - [ ] `it`;
      - [ ] `la`;
      - [ ] `pl`;
      - [ ] `pt-br`;
      - [ ] `sk`;
   - [ ] `mary_mother_of_hope` (no source for `en` translation):
      - `mary_mother_of_hope: 'Blessed Virgin Mary, Mother of Hope',`;
      - [ ] `cs`;
      - [ ] `es`;
      - [ ] `fr`;
      - [ ] `hu`;
      - [ ] `it`;
      - [ ] `la`;
      - [ ] `pl`;
      - [ ] `pt-br`;
      - [ ] `sk`;
   - [ ] `miltiades_pope` ([src](https://en.wikipedia.org/wiki/Pope_Miltiades)):
      - `miltiades_pope: 'Saint Miltiades, Pope',`;
      - [ ] `cs`;
      - [ ] `es`;
      - [ ] `fr`;
      - [ ] `hu`;
      - [ ] `it`;
      - [ ] `la`;
      - [ ] `pl`;
      - [ ] `pt-br`;
      - [ ] `sk`;
   - [ ] `our_lady_of_angels` ([src](https://en.wikipedia.org/wiki/Our_Lady_of_Angels)):
      - check where is this title of Our Lady celebrated in Greece and whether it is linked to the monastery in Croatia;
      - `our_lady_of_angels: 'Our Lady of Angels',`;
      - [ ] `cs`;
      - [ ] `es`;
      - [ ] `fr`;
      - [ ] `hu`;
      - [ ] `it`;
      - [ ] `la`;
      - [ ] `pl`;
      - [ ] `pt-br`;
      - [ ] `sk`;
   - [ ] `our_lady_of_faneromeni` (no source for `en` translation; rename to `our_lady_of_lefkada`):
      - `our_lady_of_faneromeni: 'Our Lady of Faneromeni',`;
      - [ ] `cs`;
      - [ ] `es`;
      - [ ] `fr`;
      - [ ] `hu`;
      - [ ] `it`;
      - [ ] `la`: `B. Mariæ Virginis de Faneromeni`;
      - [ ] `pl`;
      - [ ] `pt-br`;
      - [ ] `sk`;
   - [ ] `pelagia_the_penitent_virgin` ([src](https://en.wikipedia.org/wiki/Pelagia)):
      - `pelagia_the_penitent_virgin: 'Saint Pelagia the Penitent',`;
      - [ ] `cs`;
      - [ ] `es`;
      - [ ] `fr`;
      - [ ] `hu`;
      - [ ] `it`;
      - [ ] `la`;
      - [ ] `pl`;
      - [ ] `pt-br`;
      - [ ] `sk`;
   - [ ] `romanus_martyr` ([src](https://en.wikipedia.org/wiki/Romanus_Ostiarius)):
      - `romanus_martyr: 'Saint Romanus, Martyr',`;
      - [ ] `cs`;
      - [ ] `es`;
      - [ ] `fr`;
      - [ ] `hu`;
      - [ ] `it`;
      - [ ] `la`;
      - [ ] `pl`;
      - [ ] `pt-br`;
      - [ ] `sk`;
   - [ ] `thecla_of_iconium_virgin` ([src](https://en.wikipedia.org/wiki/Thecla)):
      - `thecla_of_iconium_virgin: 'Saint Thecla of Iconium, Virgin and Martyr',`;
      - [ ] `cs`;
      - [ ] `es`;
      - [ ] `fr`;
      - [ ] `hu`;
      - [ ] `it`;
      - [ ] `la`;
      - [ ] `pl`;
      - [ ] `pt-br`;
      - [ ] `sk`;
   - [ ] `theodore_stratelates` ([src](https://en.wikipedia.org/wiki/Theodore_Stratelates)):
      - `theodore_stratelates: 'Saint Theodore Stratelates, Martyr',`;
      - [ ] `cs`;
      - [ ] `es`;
      - [ ] `fr`;
      - [ ] `hu`;
      - [ ] `it`;
      - [ ] `la`;
      - [ ] `pl`;
      - [ ] `pt-br`;
      - [ ] `sk`;
   - [ ] `twelve_apostles`:
      - `twelve_apostles: 'The Holy Twelve Apostles',`;
      - [ ] `cs`;
      - [ ] `es`;
      - [ ] `fr`;
      - [ ] `hu`;
      - [ ] `it`;
      - [ ] `la`;
      - [ ] `pl`;
      - [ ] `pt-br`;
      - [ ] `sk`;
   - [ ] `ursula_of_cologne_virgin`:
      - `ursula_of_cologne_virgin: 'Saint Ursula, Virgin and Martyr',`;
      - [ ] `cs`;
      - [ ] `es`;
      - [ ] `fr`;
      - [ ] `hu`;
      - [ ] `it`;
      - [ ] `la`;
      - [ ] `pl`;
      - [ ] `pt-br`;
      - [ ] `sk`;
   - [ ] `zechariah_and_elizabeth_parents_of_john_the_baptist`:
      - `zechariah_and_elizabeth_parents_of_john_the_baptist: 'Saints Zechariah and Elizabeth, parents of John the Baptist',`;
      - capitalise `Parents` title;
      - add `Parents` title to `Titles` enum;
      - [ ] `cs`;
      - [ ] `es`;
      - [ ] `fr`;
      - [ ] `hu`;
      - [ ] `it`;
      - [ ] `la`;
      - [ ] `pl`;
      - [ ] `pt-br`;
      - [ ] `sk`;
- [ ] fix and remove `TODO` items;

</details>

emagnier commented:
Thank you both for this fabulous job! Both for this extensive localization and information addition work, as well as for the meticulous PR review 💪 🙏 

@tukusejssirs, your notes and to-do list are very comprehensive and well done, but I also find them a bit intimidating 😨. Let's be careful not to scare off or discourage contributions 😄. However, this highlights the need to create a much more comprehensive contribution guide that we can refer to, with different levels of possible contributions. 
And in general (as I've mentioned elsewhere), I prefer that we don't make PRs too large and take it step by step, depending on everyone's time and availability.

In case there are localizations that we're unsure about, but can't verify quickly, I suggest including these comments in the localization file (as code comments). While it's not ideal, at least it won't block the PR (while keeping a record of points to check).

We should also think and find a more sustainable way to specify sources for each added piece of information (to be discussed and addressed in a separate topic).

emagnier commented on lib/locales/en.ts:
```diff
@@ -639,15 +648,18 @@ export const locale: Locale = {
     luigi_orione_priest: 'Saint Luigi Orione, Priest',
     luke_evangelist: 'Saint Luke, Evangelist',
     lydia_of_philippi: 'Saint Lydia of Philippi',
+    lydia_of_philippi_first_european_christian: 'Saint Lydia of Philippi, the first Christian of Europe',
```
Note that in `lib/particular-calendars/greece.ts`, there is still no references for `lydia_of_philippi_first_european_christian`.

If you want to apply a different locale (other than the default one `lydia_of_philippi`), only and specifically for this calendar, you should modify the `lydia_of_philippi` calendar definition in `greece.ts` to:
```
lydia_of_philippi: {
  precedence: Precedences.ProperMemorial_11b,
  customLocaleId: 'lydia_of_philippi_first_european_christian',
  dateDef: { month: 8, date: 3 },
},
```

tukusejssirs replied:
Actually, `lydia_of_philippi` is inscribed in the Greek National Proper only (no other calendar in romcal has that celebration inscribed).

Therefore, I think it it safe to replace `lydia_of_philippi` with `lydia_of_philippi_the_first_christian_of_europe` (updated to what I believe is a better translation of the title) in every locale and `greek.ts`. Moreover, it would be nice to add `FIRST_CHRISTIAN_OF_EUROPE` (or `FIRST_CHRISTIAN` with place set to `EUROPE`) to the list of patronage titles.

tukusejssirs replied:
`lydia_of_philippi_first_european_christian` should be renamed to `lydia_of_philippi_the_first_christian_of_europe`.

The question is, however, whether we should also rename it in `martyrology.ts` or only add `martyrology: ['lydia_of_philippi']` in the celebration definition in `greece.ts`. We should also consider the value of `.name` in there (that depends on whether we add the new patronage title, see the previous comment of mine). :thinking:

I am about to rename it everywhere (incl `martyrology.ts` and `.name` will not be changed) to `lydia_of_philippi_the_first_christian_of_europe`.

emagnier commented on lib/locales/en.ts:
```diff
@@ -944,13 +965,17 @@ export const locale: Locale = {
     thorlac_of_iceland_bishop: 'Saint Thorlac, Bishop',
     thursday_of_the_lords_supper: 'Thursday of the Lord’s Supper',
     timothy_of_ephesus_and_titus_of_crete_bishops: 'Saints Timothy and Titus, Bishops',
+    timothy_of_ephesus_and_titus_the_first_bishop_of_crete_bishops:
+      'Saint Titus, the first Bishop of Crete, Saint Timothy, Bishops',
```
Same as for `lydia_of_philippi_first_european_christian`.

The locale `timothy_of_ephesus_and_titus_the_first_bishop_of_crete_bishops` is not referenced elsewhere other than in the locale files. As the corresponding definition is already defined in the `proper-of-saints.ts`, you should add a calendar definition in `greece.ts` that extends its definition, and specify the custom locale entry only in this particular calendar:

```ts
// To be added in greece.ts
timothy_of_ephesus_and_titus_of_crete_bishops: {
   customLocaleId: 'timothy_of_ephesus_and_titus_the_first_bishop_of_crete_bishops',
},
```

emagnier commented on lib/locales/en.ts:
```diff
@@ -944,13 +965,17 @@ export const locale: Locale = {
     translation_of_the_relics_of_odile_of_alsace_abbess: 'Translation of the Relics of Saint Odile',
     translation_of_the_relics_of_saint_stephen_of_hungary: 'Translation of the Relics of Saint Stephen of Hungary',
     turibius_of_mogrovejo_bishop: 'Saint Turibius of Mogrovejo, Bishop',
+    twelve_apostles: 'The Holy Twelve Apostles',
```
That one does not have a matching calendar definition. Is the calendar definition missing? Or is it a locale that should be applied to an existing element but only for `greece.ts`?

emagnier commented on lib/locales/en.ts:
```diff
@@ -980,6 +1005,7 @@ export const locale: Locale = {
     yolanda_of_poland_religious: 'Blessed Yolanda, Religious',
     zdenka_cecilia_schelingova_virgin: 'Blessed Zdenka Cecilia Schelingová, Virgin and Martyr',
     zdislava_of_lemberk: 'Saint Zdislava',
+    zechariah_and_elizabeth_parents_of_john_the_baptist: 'Saints Zechariah and Elizabeth, parents of John the Baptist',
```
Same here as `twelve_apostles`, there is actually no matching calendar definitions.

emagnier commented on lib/locales/en.ts:
```diff
@@ -343,13 +346,17 @@ export const locale: Locale = {
     davnet_of_sliabh_beagh_virgin: 'Saint Davnet, Virgin',
     declan_of_ardmore_bishop: 'Saint Declan, Bishop',
     dedication_of_consecrated_churches: 'The Dedication of Consecrated Churches Whose Date of Consecration is Unknown',
+    dedication_of_the_basilica_of_our_lady_of_the_snows_tinos_grece:
+      'The Dedication of the Basilica of Our Lady of the Snows in Tinos, Grece', // aka. Salus Populi Romani
```
The calendar definition for this item is missing, you should add it to `greece.ts`.

TobiTenno replied:
I was trying to resolve some of these, but I can't find references to our lady of snows in tinos yet.

there's definitely an "our lady of tinos"...

tukusejssirs replied:
This needs to be defined in `greece.tinos.ts`. It can be sourced from [this file](https://drive.google.com/file/d/1Z-ij1i4oeugS8XiIAao5iYyDHKPEvOz3/view?usp=drive_link) from p449 (5 August/ΑΥΓΟΥΣΤΟΣ).

emagnier commented on lib/locales/en.ts:
```diff
@@ -343,13 +346,17 @@ export const locale: Locale = {
     dedication_of_consecrated_churches: 'The Dedication of Consecrated Churches Whose Date of Consecration is Unknown',
+    dedication_of_the_basilica_of_our_lady_of_the_snows_tinos_grece:
+      'The Dedication of the Basilica of Our Lady of the Snows in Tinos, Grece', // aka. Salus Populi Romani
     dedication_of_the_basilica_of_saint_mary_major: 'The Dedication of the Basilica of Saint Mary Major',
     dedication_of_the_basilicas_of_saints_peter_and_paul_apostles:
       'The Dedication of the Basilicas of Saints Peter and Paul, Apostles',
     dedication_of_the_cathedral_basilica_of_saint_denis_france:
       'The Dedication of the Cathedral Basilica of Saint-Denis, France',
     dedication_of_the_cathedral_of_notre_dame_de_strasbourg_france:
       'The Dedication of the Cathedral of Notre-Dame of Strasbourg',
+    dedication_of_the_cathedral_of_saint_dionysius_athens:
+      'The Dedication of the Basilica of Saint Dionysius in Athens, Greece',
```
The calendar definition for this item is missing, you should add it to `greece.ts`.

tukusejssirs replied:
Indeed, no all celebrations are defined in `greece.ts`. That said, I believe that this celebration will be defined in `greece.athens.ts` () instead. Moreover in `el.ts`, the value might not be correct (see [this Greek Wikipedia article](https://el.wikipedia.org/wiki/%CE%9A%CE%B1%CE%B8%CE%BF%CE%BB%CE%B9%CE%BA%CF%8C%CF%82_%CE%9A%CE%B1%CE%B8%CE%B5%CE%B4%CF%81%CE%B9%CE%BA%CF%8C%CF%82_%CE%9D%CE%B1%CF%8C%CF%82_%CE%91%CE%B3%CE%AF%CE%BF%CF%85_%CE%94%CE%B9%CE%BF%CE%BD%CF%85%CF%83%CE%AF%CE%BF%CF%85_%CE%91%CF%81%CE%B5%CE%BF%CF%80%CE%B1%CE%B3%CE%AF%CF%84%CE%B7)).

Anyway, I have quickly gone through the files shared with us, but I could not find this celebrated in there. I might have missed something though. That said, the celebration should exist anyway. :wink:

tukusejssirs replied:
Actually, it can be found in [this file](https://drive.google.com/file/d/1clZroPm6Gl3zeXpjk-e64LJ9fx1tm2yV/view) on p326.

emagnier reviewed (commented):
I'm adding a few additional comments. In addition to the specific points I mentioned, I've noticed that these items (locale & martyrology items) do not have corresponding calendar definitions (likely to be added in `greece.ts`):

- `agapitus_of_palestrina_martyr`
- `christopher_of_anatolia_martyr`
- `cyriacus_martyr`
- `lawrence_justinian`
- `mammes_of_caesarea_martyr`
- `marcellinus_of_ancona_bishop`
- `mary_mother_of_hope`
- `miltiades_pope`
- `pelagia_the_penitent_virgin`
- `romanus_martyr`
- `thecla_of_iconium_virgin`
- `theodore_stratelates`

emagnier commented:
> Names of some of saints inserted to the General Calendar after 2006 were added. If found, please report missing ones.

@ciesiolka, I ran the script I developed here https://github.com/romcal/romcal/pull/461 on your PR. Here is what it flagged as missing:
```
 WARN  Missing localized 'el' items for the Proper of Time and the General Roman Calendar:
       names:
       - ash_wednesday
       - thursday_of_the_lords_supper
       - friday_of_the_passion_of_the_lord
       - holy_saturday
       - ascension_of_the_lord
       - sunday_of_the_word_of_god
       - john_of_avila_priest
       - paul_vi_pope
       - henry_ii_emperor
       - wenceslaus_i_of_bohemia_martyr
       - hedwig_of_silesia_religious
       - dedication_of_consecrated_churches
       - columban_of_luxeuil_abbot
       - lucy_of_syracuse_virgin
```

tukusejssirs commented:
@ciesiolka, I made some updates to this PR in [my fork](https://github.com/tukusejssirs/romcal/blob/romcal-459/lib/locales/el.ts) of your `greek-locale` branch, however, it seems like I have no sufficient rights on romcal repository to push my changes to your repository. Could you force-pull my changes please? Thanks in advance! :pray:

I have verified (and fixed all typos) all the keys from the GRC (_Γενικό Ρωμαϊκό εορτολόγιο_) in `mr_el_2006_ed3`, added source comments to all keys I have verified, rebased the branch on top of `dev`.

After you (or @emagnier) force-push the changes to your branch, I want to add some review comments on some (potential) issues (I don’t want to talk about them outside of the code context).

---

@emagnier, could you give me push access to romcal repository please? I wanted to push my changes into @ciesiolka’s repository, however, although he allowed us to push to his branch (`[x] Maintainers are allowed to edit this pull request.`), it seems like I can’t do it without the ability to push to romcal repository, see the [docs](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork). Thanks! :pray:

---

I hope you both are doing well! :pray:

tukusejssirs committed:
fix(el): fix some celebration names

Changes:

- added source references based on the additional pages from `mr_el_2006_ed3` shared with me;
- added the following keys based on various sources:
  - `ascension_of_the_lord`;
  - `ash_wednesday`;
  - `divine_mercy_sunday`;
  - `easter_sunday`;
  - `friday_of_the_passion_of_the_lord`;
  - `holy_saturday`;
  - `john_of_avila_priest`;
  - `paul_vi_pope`;
  - `sunday_of_the_word_of_god`;
  - `thursday_of_the_lords_supper`;
- changed the value of the following keys based on `mr_el_2006_ed3`:
  - `cycles.proper_of_time`;
  - `seasons.paschal_triduum.season`;
- replaced the following key names, as the old key names were errors:
  - `columba_of_iona_abbot` → `columban_of_luxeuil_abbot`;
  - `eric_ix_of_sweden_martyr` → `henry_ii_emperor`;
  - `hedwig_of_poland` → `hedwig_of_silesia_religious`;
- fixed a typo: `περίοδου` → `περιόδου`.

Review of romcal#459

tukusejssirs commented:
@ciesiolka, I have just pushed some new changes to my fork, see the changelog bellow.

> fix(el): fix some celebration names
> 
> Changes:
> 
> - added source references based on the additional pages from `mr_el_2006_ed3` shared with me;
> - added the following keys based on various sources:
>   - `ascension_of_the_lord`;
>   - `ash_wednesday`;
>   - `divine_mercy_sunday`;
>   - `easter_sunday`;
>   - `friday_of_the_passion_of_the_lord`;
>   - `holy_saturday`;
>   - `john_of_avila_priest`;
>   - `paul_vi_pope`;
>   - `sunday_of_the_word_of_god`;
>   - `thursday_of_the_lords_supper`;
> - changed the value of the following keys based on `mr_el_2006_ed3`:
>   - `cycles.proper_of_time`;
>   - `seasons.paschal_triduum.season`;
> - replaced the following key names, as the old key names were errors:
>   - `columba_of_iona_abbot` → `columban_of_luxeuil_abbot`;
>   - `eric_ix_of_sweden_martyr` → `henry_ii_emperor`;
>   - `hedwig_of_poland` → `hedwig_of_silesia_religious`;
> - fixed a typo: `περίοδου` → `περιόδου`.

TobiTenno commented:
@emagnier just a reminder to you and @tukusejssirs to follow up on this, I can't help much, but it would be nice to get this and some others closed out before we completely change everything

emagnier commented:
I suggest that we rebase and merge this PR, adding at least these translations (with the help of a translation tool if necessary).

```
Missing localized 'el' items for the Proper of Time and the General Roman Calendar:
  ranks:
       - optional_memorial
  names:
       - ash_wednesday
       - thursday_of_the_lords_supper
       - friday_of_the_passion_of_the_lord
       - holy_saturday
       - ascension_of_the_lord
       - sunday_of_the_word_of_god
       - john_of_avila_priest
       - paul_vi_pope
       - henry_ii_emperor
       - wenceslaus_i_of_bohemia_martyr
       - hedwig_of_silesia_religious
       - dedication_of_consecrated_churches
       - columban_of_luxeuil_abbot
       - lucy_of_syracuse_virgin
```

It's okay if not all translations are perfect, but at least we're not letting this work fall into disuse. This will facilitate future uses of Romcal and contributions in this language, especially since we have an ongoing refactor of the entire codebase.

TobiTenno committed:
fix(l10n): type typo

TobiTenno committed:
fix(costa-rica): remove tangential addition

TobiTenno commented:
@emagnier if we can resolve enough for it to be building successfully, i'm down to make a standalone issue w/ some of @tukusejssirs' TODO's for cleanups

tukusejssirs commented:
## TOC

<!-- TOC -->
* [Current status of this PR](#current-status-of-this-pr)
* [Comments on prompts to merge this PR ASAP](#comments-on-prompts-to-merge-this-pr-asap)
* [Off-topic notes](#off-topic-notes)
<!-- TOC -->

### Current status of this PR

In this PR, there are some typos and other issues. Note that the last version of this PR that I have reviewed was d579ae890cd577e7d14e9733860c08e109fa968a (i.e. the last commit @ciesiolka made). As I was not a maintainer at that time (see [this comment of mine](https://github.com/romcal/romcal/pull/459#issuecomment-1937312095)), I have updated the code in [my own fork](https://github.com/tukusejssirs/romcal/commits/romcal-459/lib/locales/el.ts) (you can see the changed I made in [here](https://github.com/tukusejssirs/romcal/compare/d579ae890cd577e7d14e9733860c08e109fa968a..romcal-459)).

I don’t consider the PR done yet. I wanted to finish it a long time ago, but due to my work/family duties, I could not devote it the time it deserves.

Now, that I am a romcal maintainer, I could merge my changes into `ciesiolka:greek-locale`.

Also note that I have forgotten to update the TODO list in [this comment of mine](https://github.com/romcal/romcal/pull/459#pullrequestreview-1640979738), however, I have updated my local copy of it. Therefore, I have just updated it with my version of it that tracks the changes I have made to my fork of @ciesiolka’s branch. Note that I have added some tasks towards myself (e.g. to add some review comments to this PR).

He’s a short version of the TODO (this short TODO won’t be updated):

- [x] I have already verified the celebrations names in l10n files based on the [files](https://drive.google.com/drive/u/5/folders/17OFxk3S4ZMPUDoNBYQzuMfIduJPJeELW) shared with us privately.
- [ ] we need to verify celebration definitions in `greece.ts`;
- [ ] we need to verify `martyrology` definitions of celebrations added in `greece.ts`;
- [ ] we need to deal with some issues I have found (I prefer doing this in this PR, although we could move [some] of the issues into an issue and fix them in a separate PR):
   - some of these issues are raised as review comments, others are defined in the TODO list in [this comment of mine](https://github.com/romcal/romcal/pull/459#pullrequestreview-1640979738);
- [ ] some all l10n and celebration metadata from the files we received from @ciesiolka related to the Greek Proper:
   - while this is not required _per se_, I still think we should not allow any new data without being properly sourced with a reference to its source (which is done, for now, using code comments);
- [ ] localise all added celebrations to all locales supported by romcal (this could be done in a separate PR); 
- [ ] we should add diocesan calendars for the Greek dioceses (this could be done in a separate PR):
   - note that no diocesan celebration names in l10n files were verified yet.

Note that I think we should create a new issue for all tasks that should be done in separate PRs so that they are not forgotten (I am not a fan of closing issues due to them being stale).

### Comments on prompts to merge this PR ASAP

> [@emagnier]
>
> I suggest that we rebase and merge this PR

I’ll do the following:

- rebase my fork;
- apply commits from @TobiTenno to @ciesiolka’s branch;
- merge the changes from my fork to @ciesiolka’s branch.

After that, I’d like to devout as much time as I can this week.

> adding at least these translations (with the help of a translation tool if necessary)

All of them are already done and properly sourced (by referencing either a first-party or third-party source) in my fork, except for `dedication_of_consecrated_churches` (not translated at all) and `john_of_avila_priest` (which was translated manually by myself).

> It's okay if not all translations are perfect, but at least we're not letting this work fall into disuse. This will facilitate future uses of Romcal and contributions in this language, especially since we have an ongoing refactor of the entire codebase.

While I am against adding _any_ new, unsourced data to romcal, I understand your POV. Recently, I’ve became slow to respond and work on romcal. Therefore, you want to finish all PRs have open for a long time.

> if we can resolve enough for it to be building successfully, i'm down to make a standalone issue w/ some of @tukusejssirs' TODO's for cleanups

@TobiTenno, wait until I merge my changes to @ciesiolka’s branch please. :pray:

### Off-topic notes

@TobiTenno, this is an example PR that is the reason I strive to use a tool to add celebration metadata, sources, calendars and l10n. The scope separation and separation of code from data (from contribution POV) is crucial. I’ve been talking about an l10n tool for a long time (see #43), now I suggest to use/create a tool add all kinds of data (it could be one or more tools actually), and that these tools would be the _only_ way to add data to romcal. The romcal data are fairly complex, and the process of their addition/modification/review/verfication is time-consuming with too many places where human error could creep in.

> It's okay if not all translations are perfect, but at least we're not letting this work fall into disuse. This will facilitate future uses of Romcal and contributions in this language, especially since we have an ongoing refactor of the entire codebase.

@emagnier, partially I agree with you, however, not using PRs (see my [off-topic notes](#off-topic-notes) below). PRs usually start as small additions (at least from the POV of new contributors), however, most of the time, they are always huge changes to multiple files and locations in romcal. Therefore, I suggest to create changes to the romcal data in an atomic way: add one thing in place at a time. For example:

- add/update a single source to the source list;
- add/update celebration metadata of a single celebration;
- add/update a single celebration in a single calendar;
- add/update a single item in a single l10n file.

This would be done use a tool (see my comment above to @TobiTenno) which could automatically create PRs every night (or week). These PRs could be auto-merged when all CI checks pass.

TobiTenno commented:
Completely understand on the desire for an accurate translation tool, the tricky bit is it without have to be integrated as part of the PR before anything is merged, so the seemingly longer task of having references for changes is the larger lift, still. 

I'll definitely hold off merging, there's plenty that was needed, like the minimums of updating the calendar to use the new l10n values. Either way, I think I kept my commits to connects from the PR, so you can now them away if needed to make merging from your fork easier

tukusejssirs commented:
> Completely understand on the desire for an accurate translation tool, the tricky bit is it without have to be integrated as part of the PR before anything is merged, so the seemingly longer task of having references for changes is the larger lift, still.

I think I get the mostly important part of this paragraph: we need to merge the (large) PRs ASAP, so that we can continue on larger changes to romcal. That makes sense, therefore, I’ll try to finish this PR and #126 ASAP, however, I cannot promise any ETA.

> I'll definitely hold off merging, there's plenty that was needed, like the minimums of updating the calendar to use the new l10n values.

Indeed there is still much needed. I go ahead and mark this PR a draft/WIP.

> Either way, I think I kept my commits to connects from the PR, so you can now them away if needed to make merging from your fork easier

I have no problem with your commits to this PR. On the contrary, I thank you for making them, as well for fixing the commit messages. :pray:

tukusejssirs commented:
Note: I won’t fix the Prettier errors until right before the PR completion, due to my preference of keeping the l10n keys and values on the same like (easier sorting, thus potential duplicates removes), and having two spaces before a line code comment that occurs on the same line as some code.

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,493 @@
+    ambrose_of_milan_bishop: 'Αγίου Αμβροσίου, επισκόπου και διδασκάλου της Εκκλησίας',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Δεκέμβριος, p42
+    andrew_apostle: 'Αγίου Ανδρέα, Απόστολου',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Νοέμβριος, p41
+    andrew_dung_lac_priest_and_companions_martyrs: 'Αγίων Ανδρέα Dung Lac, πρεσβυτέρου, και συντρόφων, μαρτύρων',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Νοέμβριος, p41
+    andrew_kim_tae_gon_priest_paul_chong_ha_sang_and_companions_martyrs: 'Αγίου Ανδρέα Kim Tae-gon, πρεσβυτέρου; Αγίου Παύλου Chong Ha-sang, και συντρόφων, μαρτύρων',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Σεπτέμβριος, p39
```
In the Greek Missal, this celebration is (at least visually; see the attached screenshot) split into two celebrations without any conjunction or comma/colon/semicolon between them. And only the second celebration is a `μνήμη`/`memorial` (the first celebration has nothing written end the end of the row). I presume in Greek, they celebrate the celebration as one (as the whole Church), however, I am not sure why they split it like this.

Anyway, I suggest to combine the celebrations into a single celebration, i.e. to use `Αγίων Ανδρέα Kim Tae-gon, πρεσβυτέρου, Παύλου Chong Ha-sang, και συντρόφων, μαρτύρων; μνήμη`; changes:

- change `Αγίου` (singular saint in Genitive) to `Αγίων` (plurarl saint in Genitive) at the beginning;
- remove `Αγίου` before the name of the second saint.

![image](https://github.com/romcal/romcal/assets/5111894/5ae19b4f-fa5c-42b4-a7e1-05bd405a602e)

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,493 @@
+    peter_julian_eymard_priest: 'Αγίου Πέτρου Ιουλιανού Eymard, πρεσβυτέρου',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Αύγουστος, p38
+    philip_and_james_apostles: 'Αγίων Φιλίππου και Ιακώβου, Αποστόλων',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Μάϊος, p35
+    philip_neri_priest: 'Αγίου Φιλίππου Neri, πρεσβυτέρου',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Μάϊος, p35
+    pius_francesco_forgione_priest: 'Αγίου Πίου de Pietrelcina, πρεσβυτέρου',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Σεπτέμβριος, p39
```
Just a note: I added `Αγίου`/`Saint` which is missing in `mr_el_2006_ed3`. I think we can safely deviate for the celebration list of the GRC, as I presume it is included where the actual liturgical texts are printed (I have no access to the entire text of `mr_el_2006_ed3`).

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,493 @@
+    athanasius_of_alexandria_bishop: 'Αγίου Αθανασίου, επισκόπου και διδασκάλου της Εκκλησίας',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Μάϊος, p35
+    augustine_of_canterbury_bishop: 'Αγίου Αυγουστίνου Cantuariensis, επισκόπου',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Μάϊος, p35
+    augustine_of_hippo_bishop: 'Αγίου Αυγουστίνου, επισκόπου και διδασκάλου της Εκκλησίας',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Αύγουστος, p38
+    augustine_zhao_rong_priest_and_companions_martyrs: 'Αγίου Αυγουστίνου Zhao Rong, πρεσβυτέρου, και συντρόφων μαρτύρων',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Ιούλιος, p37
```
While the source is official, I see some inconsistencies regarding the usage of comma before `και` (`and`): sometimes there is a comma between a title and `και` and sometimes not.

ciesiolka replied:
Please remember that there's only a handful of Catholic Churches in Greece. Liturgical books are printed in limited quantities (which makes the price of single book high) and to keep the cost relatively low they were sparing on proof checking, etc.

tukusejssirs replied:
Okay, I get it.

You seem to know the Greek Roman Catholic Churches a bit (and you also seem to have much better knowledge of Greek than me); what format would you suggest to use regarding `και` in here? While I’d like to be faithful the the spelling used in the official books, I also want to keep the celebration names consistent (per locale OFC).

tukusejssirs committed:
fix(el): fix spelling of `dedication_of_the_lateran_basilica`

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,493 @@
+    cyril_of_jerusalem_bishop: 'Αγίου Κυρίλλου των Ιεροσολύμων, επισκόπου και διδασκάλου της Εκκλησίας',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Μάρτιος, p33
+    damasus_i_pope: 'Αγίου Δαμάσου, πάπα',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Δεκέμβριος, p42
+    dedication_of_the_basilica_of_our_lady_of_the_snows_tinos_grece: 'Των Εγκαινίων της Βασιλικής της Αγίας Μαρίας, της Παναγίας των Χιόνων',
+    dedication_of_the_basilica_of_saint_mary_major: 'Τα Εγκαίνια της βασιλικής της Αγίας Μαρίας',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Αύγουστος, p38
```
This dedication celebration has `Τα` at the beginning of its name, while `dedication_of_the_basilicas_of_saints_peter_and_paul_apostles` does not. Moreover, no other dedication celebration includes the word `Εγκαίνια`; other celebrations use `Εγκαινίων`. It would be interesting what causes the inconsistency.

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,493 @@
+    nicholas_of_myra_bishop: 'Αγίου Νικολάου, επισκόπου',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Δεκέμβριος, p42
+    norbert_of_xanten_bishop: 'Αγίου Νορβέρτου, επισκόπου',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Ιούνιος, p36
+    our_lady_of_angels: 'Της Υπεραγίας Θεοτόκου Μαρίας των Αγγέλων',
+    our_lady_of_faneromeni: 'Υπεραγίας Θεοτόκου της Φανερωμένης',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Σεπτέμβριος, p39
```
`Υπεραγίας Θεοτόκου της Φανερωμένης` basically means `Our Lady Revealed`/`Revealed Our Lady`, therefore, IMHO `of Lefkada` is better translation in English, as Our Lady revealed herself at many places and Lefkada is one of them.

ciesiolka replied:
I'm not sure if I get the `Revealed` part. `Υπεραγίας Θεοτόκου της Φανερωμένης` literally means `The Most Holy Mother of God of Faneromeni` or simply `Our Lady of Faneromeni`. Faneromeni is a name of the monastery located on Lefkada island.

tukusejssirs replied:
> I'm not sure if I get the Revealed part.

I am far from good at Greek, but quick online search gives me that `Faneromeni`/`Phaneromene`/`Fanerono` (`Φανερώνω`) means (more or less) `reveal`/`revealed` ([src](https://lefkadabeaches.gr/faneromeni-monastery/)).

However, indeed, it is not a title/name attributed to Our Lady, but a name of the monastery, therefore, we could keep it in English as `Our Lady of Faneromeni`.

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,493 @@
+    sunday_of_the_word_of_god: '3η Κυριακή της κοινής περιόδου του έτους ή Κυριακής του Λόγου του Θεού',  // src: https://episkopisyrou.gr/eortasmos-tis-kyriakis-tou-logou-tou-theou/
+    sylvester_i_pope: 'Αγίου Σιλβέστρου Αʹ, πάπα',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Δεκέμβριος, p42
+    teresa_benedicta_of_the_cross_stein_virgin: 'Αγίας Θηρεσίας Βενεδίκτης a Cruce (Edith Stein), παρθένου και μάρτυρος',  // based on: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Αύγουστος, p38
+    teresa_benedicta_of_the_cross_stein_virgin_copatroness_of_europe: 'Αγίας Θηρεσίας Βενεδίκτης a Cruce (Edith Stein), παρθένου και μάρτυρος, προστάτιδας της Ευρώπης',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Αύγουστος, p38
```
Official translation with inconsistency: `παρθενομάρτυρος` vs `παρθένου και μάρτυρος`.

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,493 @@
+    thomas_becket_bishop: 'Αγίου Θωμά Becket, επισκόπου και μάρτυρος',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Δεκέμβριος, p42
+    thursday_of_the_lords_supper: 'Μεγάλη Πέμπτη στο Μυστικό δείπνο',  // src: mr_el_2006_ed3, p???
+    timothy_of_ephesus_and_titus_of_crete_bishops: 'Αγίων Τιμοθέου και Τίτου, επισκόπων',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Ιανουάριος, p31
+    timothy_of_ephesus_and_titus_the_first_bishop_of_crete_bishops: 'Αγίου Τίτου, πρώτου επισκόπου Κρήτης και Αγίου Τιμοθέου, επισκόπου',  // based on: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Ιανουάριος, p31
```
This celebration is inscribed in `greece.syriou.ts` on 26 January ([src](https://drive.google.com/file/d/1dCvYUl89ktHuByvrIv4BDs0K1EZE2jAi/view?usp=drive_link); PDF p17 / actual p19).

This celebration should be merged with `timothy_of_ephesus_and_titus_of_crete_bishops` after we generate celebration names using celebration metadata and celebration definitions, while adding `Title.FIRST_BISHOP` to the celebration name (as it is not used in the GRC).

This problem is related to patronage titles (#221, #201, #24) and to different titles used in different calendars.

For now, we will keep it as is.

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,493 @@
+    vincent_ferrer_priest: 'Αγίου Βικεντίου Ferrer, πρεσβυτέρου',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Απρίλιος, p34
+    vincent_of_saragossa_deacon: 'Αγίου Βικεντίου, διακόνου και μάρτυρος',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Ιανουάριος, p31
+    visitation_of_mary: 'Η επίσκεψη της Αειπαρθένου Μαρίας στην αγία Ελισάβετ',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Μάϊος, p35
+    wenceslaus_i_of_bohemia_martyr: 'Αγίου Βενκεσλάου, μάρτυρος',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Σεπτέμβριος, p39
```
When I search for this celebration name on the Internet, I have found his name spelled as `Βεγκεσλάου` which seems to be the right spelling (as in Modern Greek, `γκ` is pronounced as `nk` and Latin `c` is usually transliterated as `κ`/`k`). That said, `mr_el_2006_ed3` uses the spelling used in the key value.

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,493 @@
+    advent: {
+      season: 'Περίοδος της Παρουσίας',
+      weekday: '$t(weekdays:{{dow}}) της $t(ordinals:{{week}}, {"context": "abbreviation"}) Εβδομάδας της Παρουσίας',  // src: mr_el_2006_ed3, p141
+      sunday: '$t(ordinals:{{week}}) Κυριακή της Παρουσίας',  // src: mr_el_2006_ed3, p???
```
Find the page number of the source.

ciesiolka replied:
vol 1., p. 139

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,493 @@
+    },
+
+    christmas_time: {
+      season: 'Περίοδος των Χριστουγέννων',  // src: mr_el_2006_ed3, p???
```
Find the page number of the source.

ciesiolka replied:
vol 1., p. 173

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,493 @@
+    },
+
+    ordinary_time: {
+      season: 'Κοινή περίοδος του έτους',  // src: mr_el_2006_ed3, p???
```
Find the page number of the source.

ciesiolka replied:
vol 1., p. 607

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,493 @@
+    },
+
+    lent: {
+      season: 'Περίοδος της αγίας Τεσσαρακοστής',  // src: mr_el_2006_ed3, p???
```
Find the page number of the source.

ciesiolka replied:
vol 1., p. 213

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,493 @@
+    lent: {
+      season: 'Περίοδος της αγίας Τεσσαρακοστής',  // src: mr_el_2006_ed3, p???
+      weekday: '$t(weekdays:{{dow}}) της $t(ordinals:{{week}}, {"context": "abbreviation"}) Εβδομάδας της Τεσσαρακοστής',  // src: mr_el_2006_ed3, p227
+      sunday: '$t(ordinals:{{week}}) Κυριακή της Τεσσαρακοστής',  // src: mr_el_2006_ed3, p???
```
Find the page number of the source.

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,493 @@
+      weekday: '$t(weekdays:{{dow}}) της $t(ordinals:{{week}}, {"context": "abbreviation"}) Εβδομάδας της Τεσσαρακοστής',  // src: mr_el_2006_ed3, p227
+      sunday: '$t(ordinals:{{week}}) Κυριακή της Τεσσαρακοστής',  // src: mr_el_2006_ed3, p???
+      day_after_ash_wed: '$t(weekdays:{{dow}}) των Τεφρών',  // src: mr_el_2006_ed3, p219
+      holy_week_day: 'Μεγάλη $t(weekdays:{{dow}})',  // src: mr_el_2006_ed3, p???
```
Find the page number of the source.

ciesiolka replied:
i.e. `ΜΕΓΑΛΗ ΔΕΥΤΕΡΑ` (Holy Monday): vol 1., p. 293

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,493 @@
+    },
+
+    paschal_triduum: {
+      season: 'Ιερό Πασχαλινό Τριήμερο'  // src: mr_el_2006_ed3, p???
```
Find the page number of the source.

ciesiolka replied:
vol 1., p. 305

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,493 @@
+    },
+
+    easter_time: {
+      season: 'Περίοδος του Πάσχα',  // src: mr_el_2006_ed3, p???
```
Find the page number of the source.

ciesiolka replied:
vol 1., p. 499

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,493 @@
+      season: 'Περίοδος του Πάσχα',  // src: mr_el_2006_ed3, p???
+      weekday: '$t(weekdays:{{dow}}) της $t(ordinals:{{week}}, {"context": "abbreviation"}) Εβδομάδας του Πάσχα',
+      sunday: '$t(ordinals:{{week}}) Κυριακή του Πάσχα',
+      octave: '$t(weekdays:{{dow}}) της Διακαινησίμου',  // src: mr_el_2006_ed3, p???
```
Find the page number of the source.

ciesiolka replied:
i.e. `ΔΕΥΤΕΡΑ ΤΗΣ ΔΙΑΚΑΙΝΗΣΙΜΟΥ` (Monday in octave), vol 1., p. 531

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,493 @@
+  },
+
+  cycles: {
+    proper_of_time: 'Ιδιαίτερο της περιόδου',  // src: mr_el_2006_ed3, p???
```
Find the page number of the source.

ciesiolka replied:
vol 1., p. 137

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,493 @@
+
+  cycles: {
+    proper_of_time: 'Ιδιαίτερο της περιόδου',  // src: mr_el_2006_ed3, p???
+    proper_of_saints: 'Ιδιαίτερο των αγίων',  // src: mr_el_2006_ed3, p???
```
Find the page number of the source.

ciesiolka replied:
vol 2., p. 45

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,493 @@
+    demetrius_of_thessaloniki_martyr: 'Αγίου Δημητρίου, μάρτυρος',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Οκτώβριος, p40
+    denis_of_paris_bishop_and_companions_martyrs: 'Αγίου Διονυσίου, επισκόπου, και συντρόφων, μαρτύρων',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Οκτώβριος, p40
+    dionysius_the_areopagite_bishop: 'Αγίου Διονυσίου του Αρεοπαγίτου, επισκόπου και μάρτυρος',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Οκτώβριος, p40
+    divine_mercy_sunday: 'Δεύτερη Κυριακή του Πάσχα ή της θείας Ευσπλαχνίας',  // src: mr_el_2006_ed3, p???
```
Find the page number of the source.

ciesiolka replied:
vol 1., p. 543

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,493 @@
+    dionysius_the_areopagite_bishop: 'Αγίου Διονυσίου του Αρεοπαγίτου, επισκόπου και μάρτυρος',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Οκτώβριος, p40
+    divine_mercy_sunday: 'Δεύτερη Κυριακή του Πάσχα ή της θείας Ευσπλαχνίας',  // src: mr_el_2006_ed3, p???
+    dominic_de_guzman_priest: 'Αγίου Δομινίκου, πρεσβυτέρου',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Αύγουστος, p38
+    easter_sunday: 'Κυριακή του Πάσχα της Αναστάσεως του Κύριου',  // src: mr_el_2006_ed3, p???
```
Find the page number of the source.

ciesiolka replied:
vol 1., p. 499

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,493 @@
+    francis_of_assisi: 'Αγίου Φραγκίσκου της Ασσίζης',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Οκτώβριος, p40
+    francis_of_paola_hermit: 'Αγίου Φραγκίσκου de Paola, ερημίτη',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Απρίλιος, p34
+    francis_xavier_priest: 'Αγίου Φραγκίσκου Ξαβερίου, πρεσβυτέρου',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Δεκέμβριος, p42
+    friday_of_the_passion_of_the_lord: 'Μεγάλη Παρασκευή',  // src: mr_el_2006_ed3, p???
```
Find the page number of the source.

ciesiolka replied:
vol 1., p. 320

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,493 @@
+    paul_vi_pope: 'Αγίου Παύλου ΣΤʹ, Πάπα',  // src: https://el.wikipedia.org/w/index.php?title=%CE%A0%CE%AC%CF%80%CE%B1%CF%82_%CE%A0%CE%B1%CF%8D%CE%BB%CE%BF%CF%82_%CE%A3%CE%A4%CE%84&oldid=9892699
+    paulinus_of_nola_bishop: 'Αγίου Παυλίνου Nolani, επισκόπου',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Ιούνιος, p36
+    pelagia_the_penitent_virgin: 'Αγίας Πελαγίας της Μετανοούσης',
+    pentecost_sunday: 'Κυριακή της Πεντηκοστής',  // src: mr_el_2006_ed3, p???
```
Find the page number of the source.

ciesiolka replied:
vol 1., p. 599

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,493 @@
+    thomas_apostle: 'Αγίου Θωμά, Αποστόλου',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Ιούλιος, p37
+    thomas_aquinas_priest: 'Αγίου Θωμά του Ακινάτη, πρεσβυτέρου και διδασκάλου της Εκκλησίας',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Ιανουάριος, p31
+    thomas_becket_bishop: 'Αγίου Θωμά Becket, επισκόπου και μάρτυρος',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Δεκέμβριος, p42
+    thursday_of_the_lords_supper: 'Μεγάλη Πέμπτη στο Μυστικό δείπνο',  // src: mr_el_2006_ed3, p???
```
Find the page number of the source.

ciesiolka replied:
vol 1., p. 307

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,493 @@
+    genevieve_of_paris_virgin: 'Αγίας Γενεβιέβης, παρθένου',
+    george_of_lydda_martyr: 'Αγίου Γεωργίου, μάρτυρος',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Απρίλιος, p34
+    gertrude_the_great_virgin: 'Αγίας Γερτρούδης, παρθένου',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Νοέμβριος, p41
+    gregory_of_narek_abbot: 'Αγίου Γρηγορίου του Narek, μοναχού και διδασκάλου της Εκκλησίας',
```
It would be nice to find a source for this. 

Consider using `Γρηγορίου Ναρεκίτης` ([src](https://pergamos.lib.uoa.gr/uoa/dl/object/2956210/file.pdf#page=117); cf [this](https://rootsbar.ru/el/landscape-design/dialogi-grigoriya-velikogo-i-legendy-o-zagrobnoi-zhizni-v/))

Or maybe [this](https://www.ecclesia.gr/greek/press/theologia/material/2016_3_20_analekta.pdf).

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,493 @@
+    holy_family_of_jesus_mary_and_joseph: 'Της Αγίας Οικογένειας: Ιήσου, Μαρίας και Ιωσήφ',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Δεκέμβριος, p42
+    holy_guardian_angels: 'Αγίων Φυλάκων Αγγέλων',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Οκτώβριος, p40
+    holy_innocents_martyrs: 'Αγίων Αθώων Νηπίων, Μαρτύρων',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Δεκέμβριος, p42
+    holy_saturday: 'Μεγάλο Σάββατο',  // src: https://el.wikipedia.org/w/index.php?title=%CE%9C%CE%B5%CE%B3%CE%AC%CE%BB%CE%BF_%CE%A3%CE%AC%CE%B2%CE%B2%CE%B1%CF%84%CE%BF&oldid=10005382
```
It would be nice be able to source this from `mr_el_2006_ed3`.

ciesiolka replied:
vol 1., p. 338

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,493 @@
+    anthony_zaccaria_priest: 'Αγίου Αντωνίου Μαρία Ζαχαρία, πρεσβυτέρου',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Ιούλιος, p37
+    apollinaris_of_ravenna_bishop: 'Αγίου Απολλιναρίου, επισκόπου και μάρτυρος',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Ιούλιος, p37
+    ascension_of_the_lord: 'Της Αγίας Τριάδος',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Μάϊος, p35
+    ash_wednesday: 'Τετάρτη των Τεφρών',  // src: https://episkopisyrou.gr/tetarti-ton-tefron-archi-tessarakostis-2/
```
It would be nice to source this directly from `mr_el_2006_ed3`.

tukusejssirs replied:
Find the page number of the source.

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,493 @@
+  ranks: {
+    solemnity: 'πανήγυρη',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Ιανουάριος, p31
+    feast: 'εορτή',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Ιανουάριος, p31
+    memorial: 'μνήμη',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Ιανουάριος, p31
```
Add missing `ranks.optional_memorial`. Ideally, source from `mr_el_2006_ed3` (in the GIRM, 355c).

ciesiolka replied:
`προαιρετική μνήμη`

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,493 @@
+    john_i_pope: 'Αγίου Ιωάννη Αʹ, πάπα και μάρτυρος',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Μάϊος, p35
+    john_leonardi_priest: 'Αγίου Ιωάννη Leonardi, πρεσβυτέρου',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Οκτώβριος, p40
+    john_mary_vianney_priest: 'Αγίου Ιωάννη Μαρία Vianney, πρεσβυτέρου',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Αύγουστος, p38
+    john_of_avila_priest: 'Αγίου Ιωάννη της Άβιλα, πρεσβυτέρου και διδασκάλου της Εκκλησίας',  // src: manually translated by @tukusejssirs using `john_of_the_cross_priest` and `teresa_of_jesus_of_avila_virgin` from `mr_el_2006_ed3`
```
Find a better source.

tukusejssirs commented on lib/particular-calendars/greece.ts:
```diff
@@ -32,16 +32,41 @@ export class Greece extends CalendarDef {
+    mary_mother_of_the_church: {
+      precedence: Precedences.ProperMemorial_11b,
+      dateDef: { month: 5, date: 13 },
+    },
```
Is this still inscribed in the Greek National Proper when the celebration is inscribed in the GRC on Monday after Pentecost? :thinking:

tukusejssirs reviewed (changes requested):
Status update: I have reviewed (privately) all celebrations from January through September. I want to publish them after I review the rest of the celebrations inscribed to the Greek National Proper.

There are some changes I want to discuss (some of them already listed under the an older version of the [TODO list](https://github.com/romcal/romcal/pull/459#pullrequestreview-1640979738)).

tukusejssirs committed:
fix(greece): update the celebrations based on `mr_el_2006_ed3`

tukusejssirs commented:
Status report: Unfortunately, I could find little time for romcal this week.

I have verified the celebrations inscribed in the Greek National Proper (but still no diocesan celebrations were verified). There are only a few changes.

Moreover, I stopped fighting with Prettier, I fixed the formatting of the files.

---

@ciesiolka, it would be awesome if you could share with us the following pages from the Greek translation of the Roman Missal (`mr_el_2006_ed3`). :pray:

- front cover;
- title page;
- title page verso and/or the colophon page (it might be in the back matter);
- a page where we can see the celebration name of the Dedication of Consecrated Churches Whose Date of Consecration is Unknown;
- the page which includes 335c of the General Instruction of the Roman Missal which includes the Greek terms for celebration ranks (we are missing the _optional memorial_);
- a page with `holy_saturday`;
- a page with `ash_wednesday`;
- [nice to have] it would be awesome if you could tell us the page number of pages which do not have them printed on those pages (this includes some pages which you have already shared with us, and possibly some of those I request in this list);
- [nice to have] page with `pius_francesco_forgione_priest` other than the GRC list you have already shared with us, as it is missing the canonisation level.

Moreover, if you have access / could find a source for the official Greek celebration name of `gregory_of_narek_abbot` and `john_of_avila_priest`, it would be awesome.

Also, I am not sure whether `mary_mother_of_the_church` is still inscribed in the Greek National Proper on 13 May or it was removed when Pope Francis inscribed it in the GRC on Monday after Pentecost. A proof would be nice to have.

Moreover, I am not sure if you used correct diacritics on the Genetive of the month names.

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,493 @@
+    leo_i_the_great_pope: 'Αγίου Λέωντα του Μεγάλου, πάπα και διδασκάλου της Εκκλησίας',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Νοέμβριος, p41
+    louis_grignion_de_montfort_priest: 'Αγίου Λουδοβίκου Μαρία Grignion de Montfort, πρεσβυτέρου',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Απρίλιος, p34
+    louis_ix_of_france: 'Αγίου Λουδοβίκου',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Αύγουστος, p38
+    louise_de_marillac_religious: 'Αγίας Λουίζας de Marillac',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Μάρτιος, p33
```
According to the [_Liturgical Calendar of the Archdiocese of Athens_](https://drive.google.com/file/d/1clZroPm6Gl3zeXpjk-e64LJ9fx1tm2yV/view?usp=drive_link) on p44, this celebration is now spelled as `Αγίας Λουίζας ντε Μαριγιάκ`. I think we should use this (newer) spelling, as it looks like they have updated the spelling and words to the (more) Modern Greek (at least they about this in the Preamble of this above-referenced source).

ciesiolka replied:
I just want to clarify that the the preamble, indeed, mentions usage of Modern Greek but only in the refence to updated texts of Responsorial Psalms.

tukusejssirs replied:
> only in the refe[re]nce to updated texts of Responsorial Psalms.

What does it mean? Do the Greek still use the older celebration names everywhere else? That seems strange to me. :thinking:

I was under impression that they use the newer spelling everywhere, even hough the books printed before the modernisation of the celebration/saint names use the old spelling.

tukusejssirs commented on lib/locales/el.ts:
```diff
@@ -0,0 +1,493 @@
+    mary_magdalene: 'Αγίας Μαρίας της Μαγδαληνής',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Ιούλιος, p37
+    mary_magdalene_de_pazzi_virgin: 'Αγίας Μαρίας Μαγδαληνής de’ Pazzi, παρθένου',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Μάϊος, p35
+    mary_mother_of_god: 'Ογδοάδα των Χριστουγέννων: Της Υπεραγίας Θεοτόκου Μαρίας',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Ιανουάριος, p31
+    mary_mother_of_the_church: 'Tης Υπεραγίας Θεοτόκου Μαρίας, μητέρας της Εκκλησίας',  // src: mr_el_2006_ed3, Γενικό Ρωμαϊκό εορτολόγιο, Μάϊος, p35
```
According to the [_Liturgical Calendar of the Archdiocese of Athens](), p79, the new name of the celebration is `Της Υπεραγίας Θεοτόκου και Αειπαρθένου Μαρίας, Μητέρας της Εκκλησίας`.

tukusejssirs replied:
It would be nice to define when `Tης`/`Tου`/`Των` is to be included in the celebration name and when not.

ciesiolka commented:
Hi @tukusejssirs.

As I mentioned in my last e-mail to you, I had very little time to contribute to this project. However, let me fix that right now and address your requests.

I scanned pages you requested into a single PDF:
[GreekMissalSelectedPages.PDF](https://github.com/romcal/romcal/files/15448630/GreekMissalSelectedPages.PDF)

> * front cover;

Page 1 of the PDF

> * title page;

PDF, page 2

> * title page verso and/or the colophon page (it might be in the back matter);

PDF, page 3

> * a page where we can see the celebration name of the Dedication of Consecrated Churches Whose Date of Consecration is Unknown;

There's no such entry in books I have access to.

> * the page which includes 335c of
 the General Instruction of the Roman Missal which includes the Greek terms for celebration ranks (we are missing the _optional memorial_);

I bet you meant 355c. PDF, page 4

> * the page which includes 346 of the General Instruction of the Roman Missal which includes the Greek colour names; however, it does not include `gold`;

PDF, page 5

> * a page with `holy_saturday`;

PDF, page 6

> * a page with `ash_wednesday`;

PDF, page 7

> * [nice to have] it would be awesome if you could tell us the page number of pages which do not have them printed on those pages (this includes some pages which you have already shared with us, and possibly some of those I request in this list);

I'll do that later if you don't mind. Please keep in mind that Greek Missal is split into two volumes, each following own page numbering (vol. 2 starts with page 1, not 900 or so).

> * [nice to have] page with `pius_francesco_forgione_priest` other than the GRC list you have already shared with us, as it is missing the canonisation level.

It's `Αγίου Πίου de Pietrelcina, πρεσβιτέρου`, vol. 2, p. 198

> Moreover, if you have access / could find a source for the official Greek celebration name of `gregory_of_narek_abbot` and `john_of_avila_priest`, it would be awesome.

As of now I don't have such information. But I'll look for that in a spare time.

> Also, I am not sure whether `mary_mother_of_the_church` is still inscribed in the Greek National Proper on 13 May or it was removed when Pope Francis inscribed it in the GRC on Monday after Pentecost. A proof would be nice to have.

As above

> Moreover, I am not sure if you used correct diacritics on the Genetive of the month names.

Both `Ιουνίου` and `Ιούνιου` are correct; the same applies to other month names. See: [wiktionary](https://el.wiktionary.org/wiki/%CE%99%CE%BF%CF%8D%CE%BD%CE%B9%CE%BF%CF%82)

tukusejssirs commented:
Sorry, it took me longer than expected to continue working on this PR.

> I bet you meant 355c. PDF, page 4

@ciesiolka, you’re right, it was a typo. :wink:

> I'll do that later if you don't mind. Please keep in mind that Greek Missal is split into two volumes, each following own page numbering (vol. 2 starts with page 1, not 900 or so).

No problem with that. Then, I wish you could give me also the volume number for each reference used in `el.ts` (later also in `greece*.ts`, however, those references should be covered by `el.ts`). All references include `mr_el_2006_ed3` which I’d like to replace with `mr_el_2006_ed3, vol. ${volume_number}` after you provide me the information. Thanks in advance! :pray:

> It's `Αγίου Πίου de Pietrelcina, πρεσβιτέρου`, vol. 2, p. 198

I presume you have replaced `υ` with `ι` in `πρεσβιτέρου`. :wink: Thanks for the volume and page numbers! :pray:

tukusejssirs committed:
fix(el): pull `martyrology.ts` from the `dev` branch

For some reason, I have not pulled it during the rebase.
